docs/usecases/encrypted_annex.rst

```diff
@@ -0,0 +1,319 @@
+
+.. code:: bash
+
+   $ datalad save -m "Adding new file" entry-file-name.dat
+   $ datalad push --to entrystore entry-file-name.dat
+   $ datalad drop entry-file-name.dat
+
+An important technical detail about git-annex is that  ``sharedpubkey`` mode encrypts file *content* using GPG, but file *names* using `HMAC <https://en.wikipedia.org/wiki/HMAC>`_.
+However, the "HMAC cipher" (the secret used to encrypt) is stored unencrypted in the git repository.
+This makes it possible to add new files without access to the private GPG keys - but also means that
+access to the git repository will reveal file names.
+Since a RIA store combines a bare git repository with annex storage in the same location, this means that we should take care to not include sensitive information in file names.
+You can see `git-annex's documentation <https://git-annex.branchable.com/encryption/>`__ and the section :ref:`privacy` for more details.
+
+Local machine: Decryption
+-------------------------
+
+In order to retrieve the encrypted data securely from the remote server and perform processing on unencrypted data, we start once again by creating a DataLad dataset:
+
+.. code:: bash
+
+   $ datalad create derived_data
+   $ cd derived_data
+
+We then install the dataset from the RIA store on the remote server as a subdataset with input data using :command:`datalad clone` and an :term:`SSH` URL to the dataset in the RIA store.
+
+.. code:: bash
+
+   $ datalad clone -d . ria+ssh://... inputs
+
+Next, we can retrieve all data:
+
+.. code:: bash
+
+   $ datalad get inputs
+
+As long as we have the required private key, GPG will be used to quietly
+decrypt all files during the ``get`` operation, so our dataset clone
+will contain already decrypted data.
+
+At this stage we may add our data processing code (likely putting it
+under ``code`` directory, and using ``.gitattributes`` to decide whether
+code files should be tracked by :term:`Git`), and use ``datalad run`` to produce
+derived data.
+
+Since we intend all our data to be encrypted at rest also on this
+machine, we will create another set of RIA siblings and tell git-annex to use encryption.
+Because here we have access to our private key, we will use the default, more flexible, scheme with ``hybrid`` encryption keys.
+
+Note: In the ``hybrid`` mode, a private key is needed for *both* retrieval
+and deposition of annexed contents, but it is easy to add new keys
+without having to re-encrypt data.
+File content and names are encrypted with a symmetric cipher, which is itself encrypted using GPG and stored encrypted in the git repository.
+See `git-annex's documentation <https://git-annex.branchable.com/encryption/>`__ for more details.
+
+.. code:: bash
+
+   $ datalad create-sibling-ria \
+     --new-store-ok --name localstore \
+     --alias derived \
+     ria+file:///data/project/store
+   $ git annex enableremote \
+     localstore-storage \
+     keyid+=2EF857223E033AF1
```

Comment:
I think it's missing encryption type declaration here & below - may be that hybrid is the default, but even if that's the case it's probably better to be explicit
```suggestion
   $ git annex enableremote \
     localstore-storage \
     encryption='hybrid' \
     keyid+=2EF857223E033AF1
```

Reply:
Yes, I must have missed it. Fixed in 0668a40